From 636313b2b4bce8b8f6ce99d16258f10590ebf6d1 Mon Sep 17 00:00:00 2001
From: Ben Walton <bwalton@opencsw.org>
Date: Sat, 17 Sep 2011 15:14:23 +0200
Subject: [PATCH 2/5] Testsuite: Honour PATH when finding fakeroot

Some systems may have fakeroot, but not at /usr/bin.  Use PATH to see
if the tool is available.  This will allow using fakeroot on a
Solaris/OpenCSW stack where fakeroot is at /opt/csw/bin/fakeroot.

Signed-off-by: Ben Walton <bwalton@opencsw.org>
---
 testsuite/chown.test   |    6 ++++--
 testsuite/devices.test |    6 ++++--
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/testsuite/chown.test b/testsuite/chown.test
index ee1f020..5631747 100644
--- a/testsuite/chown.test
+++ b/testsuite/chown.test
@@ -37,12 +37,14 @@ case $0 in
     ;;
 *)
     RSYNC="$RSYNC --super"
+    # make fakeroot detection take path into account
+    fr="`which fakeroot 2>/dev/null`"
     case `get_testuid` in
     '') ;; # If "id" failed, try to continue...
     0)  ;;
-    *)  if [ -f /usr/bin/fakeroot ]; then
+    *)  if [ -n "${fr}" ]; then
 	    echo "Let's try re-running the script under fakeroot..."
-	    exec /usr/bin/fakeroot /bin/sh "$0"
+	    exec "${fr}" /bin/sh "$0"
 	fi
 	;;
     esac
diff --git a/testsuite/devices.test b/testsuite/devices.test
index fcd3eb0..30bc1c4 100644
--- a/testsuite/devices.test
+++ b/testsuite/devices.test
@@ -51,12 +51,14 @@ case $0 in
     esac
     ;;
 *)
+    # make fakeroot detection take path into account
+    fr="`which fakeroot 2>/dev/null`"
     case `get_testuid` in
     '') ;; # If "id" failed, try to continue...
     0)  ;;
-    *)  if [ -f /usr/bin/fakeroot ]; then
+    *)  if [ -n "${fr}" ]; then
 	    echo "Let's try re-running the script under fakeroot..."
-	    exec /usr/bin/fakeroot /bin/sh $RUNSHFLAGS "$0"
+	    exec "${fr}" /bin/sh $RUNSHFLAGS "$0"
 	fi
 	test_skipped "Rsync needs root/fakeroot for device tests"
 	;;
-- 
1.7.4.1

